Sources/PostgresNIO/New/Data/Bytes+PSQLCodable.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  File.swift
+//  
+//
+//  Created by Fabian Fett on 13.01.21.
+//
+
+import struct Foundation.Data
+import NIOFoundationCompat
+
+extension PSQLEncodable where Self: Sequence, Self.Element == UInt8 {
+    var psqlType: PSQLDataType {
+        .bytea
+    }
+    
+    func encode(into byteBuffer: inout ByteBuffer, context: PSQLEncodingContext) {
+        byteBuffer.writeBytes(self)
+    }
+}
+
+extension ByteBuffer: PSQLCodable {
+    var psqlType: PSQLDataType {
+        .bytea
+    }
+    
+    func encode(into byteBuffer: inout ByteBuffer, context: PSQLEncodingContext) {
+        var copyOfSelf = self // dirty hack
+        byteBuffer.writeBuffer(&copyOfSelf)
```

Comment:
If we use `setBuffer` we must also move the `target`'s `writerIndex`. If you don't mind I'd prefer to stay with the higher level API to prevent exactly those mistakes. ;)

Reply:
Ah, yeah - you either have to make a mutable reader index on the source buffer to move the writer index manually on the destination buffer. The way you have it is better.